lib/CAPI.cpp

```diff
@@ -9,11 +9,16 @@
 //===----------------------------------------------------------------------===//
 
 #include "CAPI.h"
+#include "Dialects/LinalgExt/Passes.h"
 
 #include "Transforms/Passes.h"
 
 using namespace mlir;
 
 void ireeLlvmSandboxRegisterPasses() {
   registerRunnersPasses();
+  linalg_ext::registerLinalgExtPasses();
+
+  // TODO: How do I get the DialectRegistry up to here?
```

Comment:
Pass a `MLIRContext` to `ireeLLlvmSandBoxRegisterPasses`, the context has its own registry. Then, on the Python side, take a defaultable context as argument with `py::none` as the default value. It will do the same trick as attrbutes/types that are created in the closest surrounding `with Context`.

Reply:
Made some progress and updated but I am still stuck with:

```
Compilation expert <python.examples.core.transform.TransformationList object at 0x7fc051ef6910>
python3: /usr/local/google/home/ntv/github/llvm-project/mlir/lib/IR/Dialect.cpp:65: void mlir::DialectRegistry::addObjectInterface(llvm::StringRef, mlir::TypeID, mlir::TypeID, mlir::ObjectInterfaceAllocatorFunction): Assertion `it != registry.end() && "adding an interface for an op from an unregistered dialect"' failed.
Aborted
```

I have been trying to register things manually to make the interface registration pass but I cannot seem to get it right.

The "automatic" global context registration and manual stuff interplay is very messy.

Reply:
Also @stellaraccident in case something pops to the trained eye.

Reply:
I don't know what you mean by automatic global context registration, there is no global context anywhere I'm aware of. You may be running into yet another instance of the infamous dynamic lib problem. Try LD_PRELOAD the sandbox .so.

Reply:
I was talking about `llvm-project/mlir/include/mlir/InitAllPasses.h`.

```

// This function may be called to register the MLIR passes with the
// global registry.
// ...
// The global registry is interesting to interact with the command-line tools.
```

But all works now thanks to your mods.

Reply:
Pass registration is different from dialect/interface.

Reply:
True, I was trying to have a single place where I can "do all registrations" and got lost; these are indeed different.